dali/operators/generic/slice/subscript.h

```diff
@@ -0,0 +1,325 @@
+  }
+
+  SubscriptArg at, lo, hi, step;
+
+  void Load(const ArgumentWorkspace &ws, int batch_size) {
+    at.Load(ws, batch_size);
+    lo.Load(ws, batch_size);
+    hi.Load(ws, batch_size);
+    step.Load(ws, batch_size);
+  }
+};
+
+
+template <typename Backend>
+class TensorSubscript : public Operator<Backend> {
+ public:
+  explicit TensorSubscript(const OpSpec &spec) : Operator<Backend>(spec) {
+    InitArgs();
+  }
+
+  USE_OPERATOR_MEMBERS();
+
+  void InitArgs() {
+    subscripts_.resize(kMaxSubscripts);
+    int num_subscripts = 0;
+    for (int i = 0; i < kMaxSubscripts; i++) {
+      SubscriptInfo &s = subscripts_[i];
+      s.Init(spec_, i);
+
+
+      if (s.IsDefined()) {
+        num_subscripts = i + 1;
+      }
+    }
+    subscripts_.resize(num_subscripts);
+  }
+
+  bool SetupImpl(vector<OutputDesc> &outputs, const workspace_t<Backend> &ws) override {
+    const auto &input = ws.template InputRef<Backend>(0);
+    DALI_ENFORCE(input.sample_dim() > 0, "Cannot apply an index to a scalar.");
+    const auto &input_shape = input.shape();
+    outputs.resize(1);
+    outputs[0].type = input.type();
+    GetRanges(ws, input_shape);
+    ProcessShapes(outputs[0].shape, input_shape);
+    return true;
+  }
+
+  void GetRanges(const workspace_t<Backend> &ws, const TensorListShape<> &in_shape) {
+    int nsub = subscripts_.size();
+    int ndim = in_shape.sample_dim();
+    DALI_ENFORCE(ndim >= nsub, "Too many indices.");
+    int nsamples = in_shape.num_samples();
+    lo_.resize(nsamples, ndim);
+    hi_ = in_shape;
+    step_.resize(nsamples, ndim);
+    for (auto &s : step_.shapes)
+      s = 1;
+    shape_ = in_shape;
+
+    for (int d = 0; d < nsub; d++) {
+      SubscriptInfo &s = subscripts_[d];
+      s.Load(ws, nsamples);
+      GetRange(s, d, in_shape);
+    }
+  }
+
+  void GetRange(SubscriptInfo &s, int d, const TensorListShape<> &in_shape) {
+    int nsamples = in_shape.num_samples();
+
+    if (s.IsIndex()) {
+      for (int i = 0; i < nsamples; i++) {
+        int64_t in_extent = in_shape.tensor_shape_span(i)[d];
+        int64_t at = s.at.values[i];
+        int64_t idx = at < 0 ? in_extent + at : at;
+        DALI_ENFORCE(idx >= 0 && idx < in_extent, make_string("Index ", idx, " is out of range "
+          "for axis ", d, " of length ", in_extent, "\n"
+          "Detected while processing sample #", i, " of shape (", in_shape[i], ")"));
+        lo_.tensor_shape_span(i)[d] = idx;
+        hi_.tensor_shape_span(i)[d] = idx + 1;
+        shape_.tensor_shape_span(i)[d] = 1;
+      }
+    }
+    if (s.IsRange()) {
+      for (int i = 0; i < nsamples; i++) {
+        int64_t in_extent = in_shape.tensor_shape_span(i)[d];
+        int64_t lo = s.lo.IsDefined() ? s.lo.values[i] : 0;
+        int64_t hi = s.hi.IsDefined() ? s.hi.values[i] : in_extent;
+        int64_t step = s.step.IsDefined() ? s.step.values[i] : 1;
+        DALI_ENFORCE(step == 1, "Indexing with non-unit step is not implemented");
```

Comment:
```suggestion
        DALI_ENFORCE(abs(step) == 1, "Indexing with non-unit step is not implemented");
```
I think negative step is allowed too (based on L220).

Reply:
No, it's not. Line 220 is future-proof, but the step is ignored, since the kernel doesn't support it yet.

Reply:
So maybe comment out L200 not to confuse the reader?

Reply:
I'd rather keep it and add strides sooner than later.

Reply:
Well, dead code is a bit misleading. I'm not asking to remove but comment out for now.

Reply:
I'll add a comment about it, but I prefer to keep the code in zombie state, but at least compiling.